spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java

```diff
@@ -0,0 +1,682 @@
+   * @param table     a table name and (optional) database
+   * @param predicate a predicate on partition columns
+   * @return matching table's partitions
+   */
+  public static List<SparkPartition> getPartitionsByFilter(SparkSession spark, String table, String predicate) {
+
+    try {
+      TableIdentifier tableIdent = spark.sessionState().sqlParser().parseTableIdentifier(table);
+      Expression unresolvedPredicateExpr = spark.sessionState().sqlParser().parseExpression(predicate);
+      Expression resolvedPredicateExpr = resolveAttrs(spark, table, unresolvedPredicateExpr);
+      return getPartitionsByFilter(spark, tableIdent, resolvedPredicateExpr);
+    } catch (ParseException e) {
+      throw ExceptionUtil.toUncheckedException(e);
+    }
+  }
+
+  /**
+   * Returns partitions that match the specified 'predicate'.
+   *
+   * @param spark         a Spark session
+   * @param tableIdent    a table identifier
+   * @param predicateExpr a predicate expression on partition columns
+   * @return matching table's partitions
+   */
+  public static List<SparkPartition> getPartitionsByFilter(
+      SparkSession spark,
+      TableIdentifier tableIdent,
+      Expression predicateExpr) {
+
+    try {
+      SessionCatalog catalog = spark.sessionState().catalog();
+      CatalogTable catalogTable = catalog.getTableMetadata(tableIdent);
+
+      Expression resolvedPredicateExpr;
+      if (!predicateExpr.resolved()) {
+        resolvedPredicateExpr = resolveAttrs(spark, tableIdent.quotedString(), predicateExpr);
+      } else {
+        resolvedPredicateExpr = predicateExpr;
+      }
+
+      return JavaConverters.seqAsJavaListConverter(
+          catalog.listPartitionsByFilter(
+              tableIdent,
+              JavaConverters
+                  .collectionAsScalaIterableConverter(ImmutableList.of(resolvedPredicateExpr))
+                  .asScala()
+                  .toSeq()))
+          .asJava()
+          .stream()
+          .map(catalogPartition -> toSparkPartition(catalogPartition, catalogTable))
+          .collect(Collectors.toList());
+    } catch (NoSuchDatabaseException | NoSuchTableException e) {
+      throw ExceptionUtil.toUncheckedException(e);
+    }
+  }
+
+  /**
+   * Returns the data files in a partition by listing the partition location.
+   *
+   * For Parquet and ORC partitions, this will read metrics from the file footer. For Avro partitions,
+   * metrics are set to null.
+   *
+   * @param partition     a partition
+   * @param conf          a serializable Hadoop conf
+   * @param metricsConfig a metrics conf
+   * @return a List of [[DataFile]]
+   */
+  public static List<DataFile> listPartition(
+      SparkPartition partition,
+      PartitionSpec spec,
+      SerializableConfiguration conf,
+      MetricsConfig metricsConfig) {
+
+    return listPartition(
+        partition.values,
+        partition.uri,
+        partition.format,
+        spec,
+        conf.get(),
+        metricsConfig);
```

Comment:
Generally, single method calls should be on a couple of lines, not an argument on each line.